### PR TITLE
V0.9

### PR DESCRIPTION
**Pull request by pchtsp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,23 +2,29 @@
...
+
 The following is a very succinct summary from the document found in `hackathonbaobab2021/data/OrganizationITC2021_V7.pdf`, which explains in detail everything about the problem, data format, etc.
 
-We want to plan one on one matches between sports team that belong to a single league (we assume there's only one league). We call the complete set of matches a tournament. This tournament follows a double round-robin. This implies that every team plays against every other team two times: once in their home stadium, the other in the adversary's. As a result we have `(N - 1) x 2` matches per team, where `N` is the total number of teams. 
+We want to plan one-on-one matches between sports team that belong to a single league (we assume there's only one league). We call the complete set of matches a tournament. This tournament follows a double round-robin. This implies that every team plays against every other team two times: once in their home stadium, the other in the adversary's. As a result we have `(N - 1) x 2` matches per team, where `N` is the total number of teams.
```
Comment (ggsdc) [inline README.md]: "...sport team**s**..."

File: hackathonbaobab2021/__init__.py [hunk 2]
```diff
@@ -23,7 +22,24 @@ class SportsScheduling(ApplicationCore):
         solvers=list(solvers.keys()),
     )
 
+    @property
+    def instance(self) -> Type[Instance]:
+        return Instance
+
+    @property
+    def solution(self) -> Type[Solution]:
+        return Solution
```
Comment (ggsdc) [inline hackathonbaobab2021/__init__.py:31]: We usually do not define this here in the `Application`, why in this case?

File: hackathonbaobab2021/core/experiment.py [hunk 3]
```diff
@@ -59,33 +72,418 @@ def from_zipped_json(
...
+                el.set(*v)
+            return el
+
+        assignment = self.solution.get_assignment().vapply(to_et)
+        for el in assignment:
+            games.append(el)
+        return games
+
+
+def compare(_dict: SuperDict[Any, int], constraints, side: Union[str, None] = "min"):
```
Comment (ggsdc) [inline hackathonbaobab2021/core/experiment.py:411]: Could the methods that are define here onwards be included in a `tools.py` file in the core folder?

File: hackathonbaobab2021/core/experiment.py [hunk 4]
```diff
@@ -59,33 +72,418 @@ def from_zipped_json(
     def solve(self, options: dict):
         raise NotImplementedError("complete this!")
 
-    def check_solution(self, list_tests: List[str] = None, **params) -> pt.SuperDict:
+    def check_solution(
```
Comment (ggsdc) [inline hackathonbaobab2021/core/experiment.py:68]: I might need some time to review the original checks in order to see if these checks perform the same operations to make sure.

File: hackathonbaobab2021/core/instance.py [hunk 5]
```diff
@@ -15,19 +15,29 @@
...
     BR2=["slots", "teams"],
     SE1=["teams"],
+    FA2=["teams", "slots"],
 )
 C_TUPLES = SuperDict(GA1=SuperDict(meetings=True))
 
 _ID = "_id"
 _CAT = "_cat"
 
+INT_PROPS = ["min", "max", "intp", "penalty"]
```
Comment (ggsdc) [inline hackathonbaobab2021/core/instance.py]: Should we move these constants as the ones in `experiment.py` to a `const.py` file in the core folder? Does it make sense?

**Review by ggsdc — CHANGES_REQUESTED** [review]
Some small comments and questions to about code structure. I might need some time to review the different checks that have to be performed better.

File: hackathonbaobab2021/__init__.py (hunk 2 above)
Comment (pchtsp) [reply hackathonbaobab2021/__init__.py:31]: I needed to have the type hinting in pycharm I think.

File: hackathonbaobab2021/core/experiment.py (hunk 3 above)
Comment (pchtsp) [reply hackathonbaobab2021/core/experiment.py:411]: they are very specific to the experiment... I guess that's why I have them there. I could move them. They could have been static methods but I think static methods take too much space inside the class.

File: hackathonbaobab2021/core/instance.py (hunk 5 above)
Comment (pchtsp) [reply hackathonbaobab2021/core/instance.py]: Yes, it makes sense. I will do that.

File: hackathonbaobab2021/__init__.py (hunk 2 above)
Comment (ggsdc) [reply hackathonbaobab2021/__init__.py:31]: Ok, I know we define it in the experiment for the same case, if you need it here no problem

File: hackathonbaobab2021/core/experiment.py (hunk 3 above)
Comment (ggsdc) [reply hackathonbaobab2021/core/experiment.py:411]: Ok, no problem

**Review by ggsdc — APPROVED** [review]

**Pull request merged**